### ISSUE TITLE
research: qualify predictive-training optimization throughput on a realistic fixture

### ISSUE THREAD

**Issue by ll7** [ll7/robot_sf_ll7#7290]
## Current issue-planning review — 2026-09-15

**Audit disposition:** `blocked`. The current audit classifies the issue as blocked; its prerequisite, evidence, or triage gate is incomplete.

**First action:** research: qualify predictive-training optimization throughput on a realistic fixture

### Current bounded plan

1. Inspect the named owners and dependencies before editing; keep the live issue scope and fail-closed evidence boundary intact.
2. Run the named focused validation, record exact output and remaining blockers, and keep fallback, degraded, unavailable, and inconclusive outcomes explicit.

### Files and owners to inspect

- No repository path was identified in the live body; establish the canonical owner before editing.

### Acceptance and validation

- Demonstrate the stated outcome from the live issue with a focused, reproducible check.
- Keep any missing prerequisite or inconclusive result explicit rather than treating it as success.

### Current dependencies and issue-text corrections

- No supplied individual draft existed for this issue in the handoff package; this is a conservative plan derived from the refreshed live body, and the historical issue text is preserved below.
- Live issue references to verify: https://github.com/ll7/robot_sf_ll7/issues/7254; https://github.com/ll7/robot_sf_ll7/issues/6482
- Keep current out-of-scope, ownership, provenance, rights, compute, and claim boundaries intact. Do not close the issue merely because this plan was posted.

### Lifecycle and claim boundary

- Current state labels: `state:blocked-external-input`.
- The issue remains open. No closure, compute submission, release/publication, evidence deletion, or scientific-claim promotion is authorized by this review.
- This review does not request a lifecycle transition; existing assignees, ownership, valid priority labels, and unrelated labels are preserved.
- Proposed non-lifecycle label additions applied with this review: `none`.

**Evidence note:** Full live issue body and complete available comment thread were reviewed during the 2026-09-15 refresh. This is planning metadata, not implementation or benchmark evidence; historical measurements remain historical.

## Current execution decision — 2026-09-08

Keep the existing three-arm comparison from #7254. Delegate dataset selection: prefer an already preserved real trajectory dataset from the canonical predictive-training workflow that matches the loader/schema and provides representative sequence and batch scale. Select it before examining throughput outcomes and freeze its URI, manifest and checksums. A synthetic fixture may test code but cannot substitute for realistic throughput qualification. Until a qualifying dataset is available, record not_available rather than a speedup claim.

Next action: The executor may locate/stage the first qualifying existing corpus, document rights and scale rationale, and run the bounded three-arm qualification on a pinned CUDA environment with durable outputs once those factual prerequisites pass. No additional dataset-choice questionnaire is needed. Retain the external-input blocker while the actual required dataset is unavailable; extend the existing runner only where necessary and keep defaults unchanged until equivalence and throughput are measured.

This decision supersedes older requests for maintainer selection within this scope. Implementation still requires the stated validation and factual prerequisites.

## Plain-language summary

Qualify the predictive-planner training optimizations on one realistic, repository-canonical dataset before anyone recommends them as a default or claims a general GPU speedup.

## Why this issue exists

The completed child [#7254](https://github.com/ll7/robot_sf_ll7/issues/7254) proved the matched FP32/loader/AMP training-signal contract on a deterministic 512-row diagnostic fixture. On the exact merged head `33e1aabbb778a3ef4cbe9dc379c0b3ced990fc9a`, the RTX 3080 result was `equivalent_smoke`, but post-warm-up throughput was control `32,500`, FP32 loader `24,106`, and AMP+loader `18,649` examples/s. The fixture is intentionally too small to support a default-performance decision.

This follow-up owns the smallest next empirical action: repeat the same comparison on one realistic predictive-planner dataset or a canonical repository dataset of representative size. It does not reopen the implementation work in [#6482](https://github.com/ll7/robot_sf_ll7/issues/6482) and does not authorize a campaign, SLURM submission, or paper claim.

## Archetype metadata

```yaml
archetype: analysis
evidence_tier: proposal
resource: gpu
claim_boundary: diagnostic training-throughput qualification only
parent: 6482
preceding_evidence: 7254
```

## Frozen comparison contract

- Preserve the three arms from #7254: FP32 control, FP32 loader optimization, and AMP with the same loader settings.
- Use one named dataset/split with a durable source manifest, SHA-256 digest, feature schema, sample/agent/horizon counts, and an explicit availability status.
- Keep the model, optimizer, seed, batch size, update budget, warm-up boundary, repeat count, and curve metrics identical across arms; vary only the declared loader/AMP settings.
- Run at least two repeats per arm on one pinned CUDA host and record effective device/loader/AMP settings, timing, update counts, peak memory, checkpoints, and exact commit/config/dataset provenance.
- Compare curves against the frozen control-dispersion envelope before inspecting throughput; classify `equivalent_no_speedup`, `equivalent_speedup_candidate`, `not_equivalent`, or `not_available`.

## Acceptance criteria

- A config-first command accepts the realistic dataset and fails closed when its manifest, digest, feature schema, CUDA device, commit, or effective arm settings do not match the declared contract.
- The result contains a compact JSON/Markdown handoff with all arm/repeat records and a reproducible command.
- Curve equivalence and throughput are reported separately; a noisy or negative throughput result remains diagnostic and does not promote defaults.
- A positive speedup is only a candidate for a later, larger validation campaign; it is not a benchmark or paper result from this issue.
- The result and decision are posted to #6482 and this issue, with raw outputs kept out of Git.

## Stop rules and limitations

- If no representative dataset is available locally or its rights/provenance are unclear, record `not_available` and stop without fabricating a fixture.
- If CUDA, a pinned host, or exact source/dataset provenance is unavailable, fail closed.
- Do not submit SLURM jobs, change defaults, alter branch protection, or make policy-quality/planner-ranking/scientific claims.

## Proposed next step

Inspect the canonical predictive training configs and dataset registry, select one representative dataset without copying raw data into Git, then extend the existing #7254 runner only as needed to consume that dataset and emit the declared result classification.

```yaml
schema: goal_autopilot_preparation.v1
repository: ll7/robot_sf_ll7
issue: 7290
source_body_sha256: e42a447a97f2525afee238d1e050f0bb99c2239db154b65af57a9a647317020f
source_comments_sha256: 
audit_schema: open_issue_contract_audit.v1
audit_digest: 3f3c51addec2c0ac8aad5ca07f84a63669ce53bf2603a0919267fd1bd198b7c6
audit_classification: human_decision
next_action: prepare_decision_envelope
authority: maintainer_or_author
execution_mode: decision
preferred_worker: MaxRunner
expected_pr_runner_label: runner:max
implementation_admitted: False
state_ready_change_proposed: False
mutation_batch: open-issues-authority-final-20260831-a
```

This packet is preparation evidence only. It never overrides live labels, exact claim state, branch state, typed dependencies, domain gates, compute authority, release authority, or scientific evidence rules.

**Comment by ll7** [comment]
## Initial readiness result: `not_available`

I inspected the clean linked worktree at merged `main` (`481711033e3b959cbc12a1a3e96bd1cc9e4f2d02`) and the canonical predictive training configs/docs.

- No repository-tracked predictive `.npz` dataset or durable dataset manifest is present.
- The documented `output/tmp/predictive_planner/datasets/*` paths are collection outputs, not available source data in this checkout.
- The only locally generated qualification fixture is the intentionally tiny #7254 synthetic fixture, which this issue explicitly must not reuse as realistic evidence.
- I did not access private/external data or submit compute because dataset rights/provenance and availability are not established.

Per the issue stop rule, this is `not_available`, not a failed speed comparison. A future owner can resume once one approved, checksum-validated canonical dataset is staged in the worktree or a durable dataset source is explicitly authorized.

**Comment by ll7** [comment]
## External-input continuation contract

The posted `not_available` result is the correct fail-closed disposition. It is not evidence that either optimization is slow, equivalent, or unsuitable on a realistic workload. No implementation or GPU run should start until one admissible dataset packet exists.

### Minimum unblock packet

A maintainer or data owner must provide all of the following without placing raw data in Git:

1. **Authority and access**: approved durable URI or staging procedure, data owner, license/usage boundary, and confirmation that the data may be used for this diagnostic comparison.
2. **Immutable identity**: dataset name/version, split name, archive or per-file SHA-256 digest, byte size, and a signed/committed manifest. Mutable directory paths or “latest” aliases are insufficient.
3. **Schema**: tensor/array names, dtypes, shapes, feature order and units, target definition, missing-value encoding, normalization/preprocessing version, and the exact loader entry point.
4. **Scale and representativeness**: sample count, trajectory/agent count, history and prediction horizons, scene/source composition, and a factual rationale for why the workload exercises the predictive-planner training path. “Larger than the synthetic fixture” alone is insufficient.
5. **Selection independence**: evidence that this dataset/split was selected before inspecting the three-arm throughput outcome. Do not choose among available datasets based on which produces a speedup.
6. **Execution environment**: authorized pinned CUDA host, GPU model, driver/CUDA/PyTorch versions, environment-lock digest, available storage, and durable output destination.

The packet should be attached or linked in this issue and reviewed before changing the runner. Missing rights, digest, schema, or representativeness keeps the issue blocked.

### Execution plan after unblock

1. Stage the dataset read-only and verify every digest before parsing it. Emit `not_available` or `input_mismatch`; never regenerate, truncate, resample, or substitute a fixture silently.
2. Freeze the exact #7254 model, optimizer, seed, update count, batch size, checkpoint rules, curve-equivalence envelope, and arm deltas in a new preregistration manifest. Any unavoidable dataset-driven change must apply identically to all arms and be declared before outcomes.
3. Pre-register arm order. Counterbalance or deterministically randomize the order across repeats to reduce warm-cache and thermal bias. Record the generated schedule before execution.
4. Separate loader initialization, warm-up, and measured steady-state updates. Synchronize CUDA around timed regions, record data-wait and compute time separately where the existing runner supports it, and reset/record peak memory per arm.
5. Run the frozen minimum repeats. With only two repeats per arm, report the individual values and dispersion; do not make a robust speedup recommendation from a favorable mean alone.
6. Evaluate curve equivalence first using the frozen envelope. Only equivalent arms enter the throughput comparison. Keep OOM, NaN, fallback, early termination, incomplete update count, thermal/power anomaly, and effective-setting mismatch as explicit non-equivalent or invalid-run states.
7. Write one immutable result bundle containing manifest, environment, schedule, raw timing records, curves/checkpoints as allowed, stdout/stderr, and hashes. Generate the JSON/Markdown handoff from that bundle without manual value transcription.

### Decision boundary

- `equivalent_no_speedup`: valid negative diagnostic evidence; no default change.
- `equivalent_speedup_candidate`: only authorizes a separately preregistered broader validation issue.
- `not_equivalent`: blocks any performance recommendation even when timing is faster.
- `not_available` / `input_mismatch`: no comparative conclusion.

Do not reopen #7254, reinterpret its small-fixture regression as realistic performance evidence, or create a synthetic “representative” dataset to unblock this issue.

**Comment by ll7** [comment]
ll7/robot_sf_ll7#7290: accept-not-available-and-wait-for-admissible-dataset

Accept `not_available` as the correct current result and stop implementation or GPU execution.
The repository has no tracked realistic predictive dataset or durable admissible dataset packet;
the only available fixture is intentionally too small for this qualification. Absence of input is
not evidence of speedup, slowdown, equivalence, or optimization quality.

Keep the issue open but externally blocked until a data owner supplies an approved durable URI or
staging procedure, rights/usage boundary, immutable digest and manifest, full feature schema and
loader, representative scale rationale, pre-outcome selection record, pinned CUDA environment,
and durable output destination. After review, run only the frozen three-arm diagnostic with
counterbalanced order and separate curve-equivalence and throughput decisions.

Do not synthesize a “realistic” replacement, inspect outcomes to choose a dataset, submit Slurm,
change defaults, or make a policy, benchmark, planner-ranking, or paper-facing claim.

**Comment by ll7** [comment]
ll7/robot_sf_ll7#7290: accept-not-available-and-wait-for-admissible-dataset

Reaffirm `not_available` as the correct current result. No tracked realistic predictive dataset or durable admissible dataset packet exists, and the small fixture cannot support a realistic-throughput or general GPU-speedup claim.

Keep implementation and GPU execution stopped. Unblock only when a checksum-bound, rights-cleared, repository-canonical dataset packet with a reproducible hydration route is available. Do not substitute a synthetic “representative” dataset or reinterpret the small-fixture regression.

Reopen only on that admissible input or a separately approved change to the qualification question.

**Comment by ll7** [comment]
ll7/robot_sf_ll7#7290: use-first-admissible-real-dataset-for-three-arm-qualification

Delegated maintainer decision — 2026-09-08. The maintainer explicitly authorized the agent to choose and record the decision for this issue.

Keep the existing three-arm comparison from #7254. Delegate dataset selection: prefer an already preserved real trajectory dataset from the canonical predictive-training workflow that matches the loader/schema and provides representative sequence and batch scale. Select it before examining throughput outcomes and freeze its URI, manifest and checksums. A synthetic fixture may test code but cannot substitute for realistic throughput qualification. Until a qualifying dataset is available, record not_available rather than a speedup claim.

Next action: The executor may locate/stage the first qualifying existing corpus, document rights and scale rationale, and run the bounded three-arm qualification on a pinned CUDA environment with durable outputs once those factual prerequisites pass. No additional dataset-choice questionnaire is needed. Retain the external-input blocker while the actual required dataset is unavailable; extend the existing runner only where necessary and keep defaults unchanged until equivalence and throughput are measured.

Reopen only if no available real dataset can satisfy the feature/scale contract or equivalence measurements invalidate the proposed optimization.
